File: c++/CommandPattern/CookFishCommand.cpp
#include "stdafx.h"
#include "CookFishCommand.h"


CookFishCommand::CookFishCommand(Chef chef)
{
	this->chef = chef;
	Command::Name = "Ë®ÖóÓã";
}



void CookFishCommand::Cook()
{
	this->chef.CookFish();
}

File: csharp/DesignPattern.Command/CookFishCommand.cs

namespace DesignPattern.Command
{
    /// <summary>
    /// 水煮鱼
    /// </summary>
    public class CookFishCommand : Command
    {
        private Chef chef;

        public CookFishCommand(Chef chef)
        {
            this.chef = chef;
            base.Name = "水煮鱼";
        }

        public override void Cook()
        {
            this.chef.CookFish();
        }
    }
}
